src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd.MM.yyyy");
+    private Date fromDate;
+    private Date toDate;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        fromDate = getDateFromString(dateFrom);
+        toDate = getDateFromString(dateTo);
+
+        Map<String, Integer> nameSalaryMap = new HashMap<>();
+        for (String name : names) {
+            nameSalaryMap.put(name, 0);
+        }
+
+        for (String oneDataString : data) {
+            String[] dataSplitFormat = oneDataString.split(" ");
```

Comment:
All magic numbers, strings, and symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
```suggestion
            String[] dataSplitFormat = oneDataString.split(SEPARATOR);
```